src/main/java/slimeknights/tconstruct/gadgets/entity/shuriken/ShurikenEntityBase.java

```diff
@@ -0,0 +1,73 @@
+package slimeknights.tconstruct.gadgets.entity.shuriken;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.ProjectileItemEntity;
+import net.minecraft.network.IPacket;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.EntityRayTraceResult;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.registry.IEntityAdditionalSpawnData;
+import net.minecraftforge.fml.network.NetworkHooks;
+
+import javax.annotation.Nonnull;
+
+public abstract class ShurikenEntityBase extends ProjectileItemEntity implements IEntityAdditionalSpawnData {
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, World worldIn) {
+    super(type, worldIn);
+  }
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, double x, double y, double z, World worldIn) {
+    super(type, x, y, z, worldIn);
+  }
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, LivingEntity livingEntityIn, World worldIn) {
+    super(type, livingEntityIn, worldIn);
+  }
+
+    /**
+   * Get damage dealt by Shuriken
+   * Should be <= 20.0F
+   * @return float damage
+   */
+  public abstract float getDamage();
+
+  /**
+   * Get knockback dealt by Shuriken
+   * Should be <= 1.0F, Minecraft
+   * typically uses values from 0.2F-0.6F
+   * @return float knockback
+   */
+  public abstract float getKnockback();
+
+  @Override
+  protected void onEntityHit(EntityRayTraceResult result) {
```

Comment:
Entity does not die on first impact. You want to override `onImpact` like `SnowballEntity` did. I suggest also overriding `onBlockHit` to drop this as an item

Reply:
Fixed it to drop as an item now - and in all of my testing, the entity did die on first impact, so let me know if that was fixed as well.